lxd/firewall/drivers/drivers_nftables.go

```diff
@@ -0,0 +1,179 @@
+package drivers
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	deviceConfig "github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/shared"
+)
+
+const nftablesNamespace = "lxd"
+const nftableContentTemplate = "nftablesContent"
+
+// Nftables is an implmentation of LXD firewall using nftables.
+type Nftables struct{}
+
+// getIPFamily converts IP version number into family name used by nftables.
+func (d Nftables) getIPFamily(ipVersion uint) (string, error) {
+	if ipVersion == 4 {
+		return "ip", nil
+	} else if ipVersion == 6 {
+		return "ip6", nil
+	}
+
+	return "", fmt.Errorf("Invalid IP version")
+}
+
+// NetworkSetupForwardingPolicy allows forwarding dependent on boolean argument
+func (d Nftables) NetworkSetupForwardingPolicy(networkName string, ipVersion uint, allow bool) error {
+	action := "reject"
+	if allow {
+		action = "accept"
+	}
+
+	family, err := d.getIPFamily(ipVersion)
+	if err != nil {
+		return err
+	}
+
+	config := &strings.Builder{}
+	tplFields := map[string]interface{}{
+		"namespace":   nftablesNamespace,
+		"networkName": networkName,
+		"family":      family,
+		"action":      action,
+	}
+
+	nftablesCommonTable.AddParseTree(nftableContentTemplate, nftablesNetForwardingPolicy.Tree)
```

Comment:
@stgraber thought you may be interested in this part, I discovered this today. I wanted to define a common top level template of:

```
table {{.family}} {{.namespace}} {
	{{- template "nftablesContent" . -}}
}
```

This template defines the lxd table and then includes the remaining content inside it. However I needed  a way of importing different templates under the same generic name of `nftablesContent`. This is achieved using the `AddParseTree` function that adds each specific template under the generic name.